### ISSUE TITLE
 No more NG5000 "Hydration was enabled with unsupported Zone.js instance."

### ISSUE THREAD

**Issue by Jordan-Hall** [sp90/treaty#3]
I've managed to find a work around the NG5000 issue that also disables unsupported Zone.js instance. Adding this to main.ts it can be added to pollyfill.ts  it a mini zone that does no patching and just works. Ensuring key isAngularZone is true resolves issues related to noticing its unsupported version of zonejs

```typescript
type Zone = {
  current: {
    get(key: string): boolean | undefined;
    run<T>(fn: (...args: any[]) => T, applyThis?: any, applyArgs?: any): T;
  };
  run<T>(fn: (...args: any[]) => T, applyThis?: any, applyArgs?: any): T ;
};
declare var Zone:  Zone
declare module globalThis {
	let Zone: Zone
}

if (typeof Zone === 'undefined') {
  const mockZone = {
    current: {
      
      get: (key: string): boolean | undefined => key === 'isAngularZone' ? true : undefined,
      run<T>(fn: (...args: any[]) => T, applyThis?: any, applyArgs?: any): T {
        return fn.apply(applyThis, applyArgs);
      }
    },
    run<T>(fn: (...args: any[]) => T, applyThis?: any, applyArgs?: any): T {
      return fn.apply(applyThis, applyArgs);
    }
  };
  globalThis.Zone = mockZone;
}
```

**Comment by sp90** [comment]
Ran the server.ts after the merge, and it didn't solve our hydration problem nor the error in the terminal, feel free to give it another go, will spend some time looking into the hydration for now and ignore the error at least

<img width="1527" alt="Screenshot 2024-02-11 at 20 09 15" src="https://github.com/sp90/treaty/assets/3041602/a440446a-b3a9-47fc-8d67-db94cc58044f">

**Comment by Jordan-Hall** [comment]
Thays due to the required input now :( running sooner than expected. Why I said it solves the first part thay also turns it off. They likely other bits that need adding. It was just first steps

**Comment by sp90** [comment]
Fair - Anyhow the NG5000 persists so I will keep the issue here for tracking

**Comment by Jordan-Hall** [comment]
Thays fine I was surprised it was closed. I noticed the other when after. I think try make the HTTp into Client not eden. I'm working on a angular Eden type anyway. It may help with the hydration and the error for the input and http to be done on server

**Comment by sp90** [comment]
Yeah, I think it might be because I disabled the whole building of SSR and just served the server with bun to bypass that part of the build i suspect that step carries some logic that is injected into the server which might be the key to solving this.

**Comment by sp90** [comment]
@Jordan-Hall you where right its the new WritableSignal inputs that are the issue if i fallback to the old input the SSR works like a charm

The NG05000 is still there but it does render the API response

**Comment by Jordan-Hall** [comment]
Without zones the onInit doesn't get called which affects the computed. I think we need to create a zone area or look at it. I did always say the old way has so many more benefits 😅 maybe use old with setter. On change it makes the HTTP call

**Comment by Jordan-Hall** [comment]
I think what you see now is a warning compared to before which thrown an error

**Comment by sp90** [comment]
I would love if we could "applySignalInputs" ourselves on component render

**Comment by Jordan-Hall** [comment]
What order would you prefer. Typed httpClient client worked from the server app type like Eden does it. Or looking at zoneless the apply thr input?

**Comment by sp90** [comment]
I guess a typed httpClient, on the other hand, we could use eden or do you think there is a benefit to using httpClient?

**Comment by Jordan-Hall** [comment]
Eden is just out the box typed for fetch and they dot notion. Using http client and typing how Eden does it, you get zero breaking changes, support for interceptors and built in http hydration and caching from my understanding

**Comment by sp90** [comment]
Eden has an config option

```
client = edenTreaty<App>('http://localhost:4201', {
    fetcher: // expects typeof fetch
  }).api;
```
So I guess if you could wrap httpClient to take that 

Yeah or as you suggested a typed httpClient, or a wrapper library using the types to result something similar to eden but using httpClient as core and with an improved interface i guess

**Comment by Jordan-Hall** [comment]
My thinking is a wrapper lib called EdenAngular or EdenClient. I was looking at using eden fetch under the hood, but not sure if it would gain anything in itself.  We could extend httpBackend to use edenFetch. But we would still want a strongly typed client. So would then do what eden fetch does. Create a wrapper around httpclient

**Comment by sp90** [comment]
True! 

I personally like the wrapper lib around httpClient the most even though it carries a bit more weight, as a V1 at least

**Comment by Jordan-Hall** [comment]
I'll get to work on that. Then can always come back to the remaining signal input issue with zoneless

**Comment by sp90** [comment]
Yeah i was thinking maybe to leave it hanging and wait for the angular team to support it tbh

**Comment by Jordan-Hall** [comment]
I'm not sure if this can be closed after #5 & #6  is merged because it has hydration

**Comment by sp90** [comment]
I will validate after I'm done merging
